holmes/core/tool_calling_llm.py

```diff
@@ -45,16 +49,22 @@ class ToolCallResult(BaseModel):
     tool_call_id: str
     tool_name: str
     description: str
-    result: StructuredToolResult
+    result: Union[StructuredToolResult, str]
     size: Optional[int] = None
 
     def as_dict(self):
+        # TODO: remove this after FE is ready
+        if isinstance(self.result, StructuredToolResult):
+            result = self.result.model_dump()
```

Comment:
Shouldn't this use something like the function below? ToolCallResult.result use to be a string and `model_dump()` returns a dict.

```python
def format_structured_tool_result(tool_call_result:StructuredToolResult):
   tool_response = tool_call_result.result.data
   if tool_call_result.result.status == ToolResultStatus.ERROR:
     tool_response = f"{tool_call_result.result.error or 'Tool execution failed'}:\n\n{tool_call_result.result.data or ''}".strip()
   return tool_call_result
```

Reply:
Well, this is actually the same as `_get_formatted_tool_call_response_content` defined below. `_get_formatted_tool_call_response_content` does not need to be a class method but can be an independent function which then allows you to use it here.